src/metabase/driver/sql_jdbc/sync/interface.clj

```diff
@@ -32,8 +35,14 @@
 (defmulti syncable-schemas
   "Return a reducible sequence of string names of schemas that should be synced for the given database. Schemas for
   which the current DB user has no `SELECT` permissions should be filtered out. The default implementation will fetch
-  a sequence of all schema names from the JDBC database metadata and filter out any schemas in `excluded-schemas`."
-  {:added "0.39.0", :arglists '([driver ^java.sql.Connection connection ^java.sql.DatabaseMetaData metadata])}
+  a sequence of all schema names from the JDBC database metadata and filter out any schemas in `excluded-schemas`, along
+  with any that shouldn't be included based on the given inclusion and exclusion patterns (see the
+  `metabase.driver.sync` namespace for full explanation)."
+  {:added "0.39.0", :arglists '([driver
+                                 ^java.sql.Connection connection
+                                 ^java.sql.DatabaseMetaData metadata
+                                 ^String schema-inclusion-patterns
+                                 ^String schema-exclusion-patterns])}
```

Comment:
What happens to all the drivers that implemented the 3-arity version of this? 💀 

Kind of annoying but if we want to be responsible and developer-friendly we really ought to just make this a new method like we've done in similar situations in the past.

At any rate, changes also need to be mentioned in https://github.com/metabase/metabase/blob/master/docs/developers-guide/driver-changelog.md... since the file actually exists now it's probably best to do it as part of the PR so it can be reviewed together with the other changes

Reply:
Fixed this, if you can have a look at the update.